assets/src/components/form/validations/index.js

```diff
@@ -0,0 +1,194 @@
+/**
```

Comment:
Many of these validations don't appear to be specific to forms.   I think the location many of these helper function could be moved.  Maybe in `data/helpers/` (which would get exported to `eejs.helpers` and is aliased to `@eventespresso/helpers`)?

Also, you may consider doing the following:

- put boolean check type functions into `booleans.js`
- put sanitization check type functions into `sanitize.js`
etc.

Reply:
Also, these kind of low-level purpose functions are useful to have tests for before review.  I understand holding off on doing automated tests for things like the form components etc but tests here will verify that your logic is sound, something that a review wouldn't pick up and thus mean an initial review has to be done again (perhaps unnecessarily).

Reply:
Also, I'd really like to see more description in the jsdocs for these type of functions.  I can read the code and get an idea for what its intended purpose is, but without tests demonstrating expectations its hard to know for certain whether the code is doing what you intend it to do - so I'm just evaluating based on the code.

Reply:
> Many of these validations don't appear to be specific to forms.

How so? All of them either return an error string if the validation fails, or `null`. That seems pretty specific to forms, although I guess it can be argued that those same error strings could be applied to other scenarios. 

FYI: the format used for these is what is required by `React-Final-Form` 

> put boolean check type functions into booleans.js

sry... but which of these do you consider boolean checks? Do you mean ones that test for a boolean value (of which there are none) ? Or do you mean ones that use an `is*` naming schema like `isNumeric` ?

> put sanitization check type functions into sanitize.js

none of these are performing sanitization in that they do not return a mutated value. If the supplied value passes the validation check, then the original value is allowed for that input.
I do think we will need sanitization functions (like running integers through `parseInt()` prior to persistence) but I have yet to get to that point with this and am not certain that that should be part of the forms layer anyways. That should likely happen at the model level prior to updates or inserts no?

> these kind of low-level purpose functions are useful to have tests for before review.

ya I don't see why... what would it matter if some tests proved a function worked if you disagreed fundamentally with the structure and/or design and/or some other aspect of that function? For example... if unit tests showed that the `isFloat` function below correctly identified floats versus other data types... but you completely disagreed with its structure/design/whatever then what does the existence of the unit test matter? 

You in fact proposed a completely different approach for the `isFloat` function below (which I'm not criticizing at all and I'm always open to suggestions for better ways to do things, which you very often provide) but would that have been any different if there was a unit test for the existing `isFloat` function? Would the existence of a unit test make your suggestion any less valid? I don't think so.

I'm not trying to be a PITA... I know we totally disagree about this point, I'm just trying to illustrate my side of the argument and show that the existence of tests beforehand has no bearing whatsoever on the code review process (imho)

> I'd really like to see more description in the jsdocs for these type of functions.

sure I can add some more jsdoc descriptions.

Reply:
> FYI: the format used for these is what is required by React-Final-Form

Had no idea.  That does provide additional important context :)

> sry... but which of these do you consider boolean checks? Do you mean ones that test for a boolean value (of which there are none) ? Or do you mean ones that use an is* naming schema like isNumeric ?
> none of these are performing sanitization in that they do not return a mutated value. If the supplied value passes the validation check, then the original value is allowed for that input.
I do think we will need sanitization functions (like running integers through parseInt() prior to persistence) but I have yet to get to that point with this and am not certain that that should be part of the forms layer anyways. That should likely happen at the model level prior to updates or inserts no?

Sorry its not immediately clear from what I wrote but mostly I was giving examples of how the code could be split up into different files for those various purposes.  I wasn't saying that what you had written here would naturally fit into all those split up files.  I just want us to be aware/reminded of how we can organize things for _reusability_.  

re unit tests:

I'm not trying to be a PITA either but I do struggle with the pushback on writing them.  As I said I get holding off writing tests for complicated things, but you're going to have to write tests anyways so why not get them done.  Aside from helping to demonstrate the expected behaviour of code, it also is very useful if you have to refactor your code because of code review or because of changes you realize you need to make as you begin user testing things.  Those pre-written tests help point out things that get broken in a refactor either because the expectations for the test change OR you broke something in dependent code unexpectedly.  Case in point, after I finished the first pass of the model-entity work I did, I wrote tests for everything.  Then I started user testing and realized I had to rework some of the code I had done.  Some of my tests broke because of expectation changes, but I had tests covering what I thought was a completely unrelated thing that broke that were NOT expectation change but revealed an actual flaw with the original code I had written.  Without those tests I probably wouldn't have caught it as quickly.  So I will continue to argue for writing tests even before submitting reviews and you can continue to ignore that if you want, that's your prerogative.